src/core/imap.ts

```diff
@@ -132,10 +193,11 @@ export class IMAP {
     }
 
     monitor() {
-        let setupTempFolder: SetupTempFolder = null;
+        const setupTempFolder: SetupTempFolder = null;
         this.monitorId = setInterval(async () => {
-            if (this.query) {
-                const {mailBox, criteria} = this.query;
+            if ((this.fromMonitoring || this.mailBoxMonitoring) && !this.pending) {
+                const {mailBox, criteria, folderId} = this.switcher^1 && this.fromMonitoring ? this.fromMonitoring : this.mailBoxMonitoring || this.fromMonitoring;
+                this.switcher ^= 1;
```

Comment:
what does this do? the logic here isn't very clear

Reply:
When searching for messages, you must first select a mailbox and then fetch messages.

But now we have two queries: first `emailMonitoring` based on inbox and second `mailBoxMonitoring`  chosen by the user, so what I do is use xor to check if there are new messages from the first query and in the second interval, to check if there are new messages from the second query, and so on.
0 1 0 1 0 1 0 1

![first](https://user-images.githubusercontent.com/58605547/188290725-3f4ab12d-0a43-40ee-b1cc-c3e30ca68768.png)